test/libsolidity/syntaxTests/iceRegressionTests/storage_parameter_in_library.sol

```diff
@@ -0,0 +1,8 @@
+library Test {
+    struct Nested { mapping(uint => uint )[2][] a; } uint y;
+    struct X { Nested n; }
+    function f(X storage x){}
+}
+// ----
+// SyntaxError 4937: (107-132): No visibility specified. Did you intend to add "public"?
+// TypeError 9957: (68-74): Library cannot have non-constant state variables
```

Comment:
Why not have a test that doesn't have any errors?

```solidity
  struct Nested { mapping(uint => uint )[2][] a; } 
  struct X { Nested n; }
  library Test {
	  function f(X storage x) public {}
  }  
```

Reply:
Yeah. I'd even say that it's more relevant than the broken example and we could keep just this one.

Initially I did not notice that the `uint y;` part is just a storage variable declaration (that would be valid in a contract) and not some invalid syntax that breaks the struct declaration. If the error happens on valid code too then the errors are just an extra, unnecessary complication.

Reply:
I'd also add tests where it's not a storage parameter and where it's not a library function (unless we already have ones covering this, which is likely).

Reply:
> Yeah. I'd even say that it's more relevant than the broken example and we could keep just this one.

There is no guarantee that the simplified test succeeds/fails iff the originally reported issue (with fixes) succeeds/fails. So keeping both tests might be considered.

> I'd also add tests where it's not a storage parameter and where it's not a library function (unless we already have ones covering this, which is likely).

I hope non-storage parameters are covered for libraries and contracts both (see e.g.
**types\mapping\library_nested_mapping.sol**,
**nameAndTypeResolution\043_functions_with_structs_of_non_external_types_in_interface_nested.sol**).

Other tests are added to **syntaxTests/types/mapping**, because it seems to fit to all of them.

Reply:
You can never be sure but I think that in this case the likelihood is very low. You just can't test everything :)

I think I'd personally just skip that test case but I'm not completely opposed to including it in `iceRegressionTests/` if you think it adds some value. More tests is usually a good thing.